### PR TITLE
Drop samples with a null byte in their label values.

### PR DESCRIPTION
**Pull request by tomwilkie**
This was causing a prod outage (as we used MustEncode to build the chunk range key, which panics on null bytes). 

We'll drop samples for now, and in the future figure out a better way of not dropping the data.

### CODE REVIEW

**Comment by jml** [comment]
Relates to weaveworks/service-conf#507

File: ingester/ingester.go [hunk 1]
```diff
@@ -259,6 +260,16 @@ func (i *Ingester) Append(ctx context.Context, samples []*model.Sample) error {
 }
 
 func (i *Ingester) append(ctx context.Context, sample *model.Sample) error {
+	userID, err := user.GetID(ctx)
+	if err != nil {
+		return err
+	}
+
+	if err := util.ValidateSample(sample); err != nil {
+		log.Errorf("Error validating sample from user '%s': %v", userID, err)
```
Comment (jml) [inline ingester/ingester.go]: Maybe get the `userID` inside this `if` block?

File: util/validate.go [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+	"regexp"
+	"strings"
+
+	"github.com/prometheus/common/model"
+)
+
+var validLabelRE = regexp.MustCompile(`^[a-zA-Z_][a-zA-Z0-9_]*$`)
+
+// ValidateSample returns an err if the sample is invalid
+func ValidateSample(s *model.Sample) error {
```
Comment (jml) [inline util/validate.go]: Danger of this approach is that we need to remember to call `ValidateSample` before handling any incoming samples. A more robust solution would be to return a validated sample that has different type to the not-yet-validated sample.

However, not necessary for this PR, and maybe not ever.

**Comment by tomwilkie** [comment]
This has been deployed to prod.  I'll go ahead and rebase into master tomorrow.

**Comment by tomwilkie** [comment]
Mitigation for #210

**Pull request merged**